### PR TITLE
Add UI, rule content choosing and content functionality service to schedule page

### LINKED ISSUE #89: Scheduler: add UI and content functionality (service)
### Describe the feature

We are currently capable of choosing both group and template within our scheduler to then send it to the desired monitors, however we cannot choose the content itself, so this issue aims to fix that.

### Use Case

When I am in the scheduler page I want to be capable of choosing not only the `Group`, `Template` but also the `Content` for each individual widget.

### Proposed Solution

UI tweaks and enhancements along with creating service to send information.

### Other Information

_No response_

### PR DESCRIPTION
**Pull request by pdPinho**
### Issue

Closes #89 

### Reason for this change

As a user I want to be capable of choosing a group and create a rule for it.
To do that I want to be capable of selecting what time content passes, what content is displayed, when it is displayed and where it is displayed (visually).
Previously none of these things were possible. We simply had two basic buttons that allowed us to send some hard-coded functionality.

### Description of changes

- Changed UI layout and built entire skeleton;
- Created two main buttons to choose group and create rule (which opens a modal);
- Created modal layout for rules;
- Added two buttons to select templates and default rules;
- Added start time, end time, start date, end date and weekdays;
- Added two default rules;
- Added button to submit and go back (submit button only becomes available under certain conditions);
- Added dynamic wireframe based on chosen template;
- Added functionality to choose content within wireframe;
- Created modal for content selection;
- Added data table to modal with existing files and folders;
- Added functionality to select, open and submit folder and files.

### Description of how you validated changes

Tested the workflow of adding content and everything else.
Made performance comparisons to guarantee that things were running smoothly.

### Checklist
- [X] I have tested my changes locally.
- [ ] I have updated the documentation to reflect my changes.
- [ ] I have added/updated unit tests (if applicable).
- [X] My code adheres to the [CONTRIBUTING GUIDE](https://ua-smart-signage-platform.github.io/Documentation-Website/guidelines/)

### Aditional Information
There are still some minor things that require revising such as:
- Information icon not displaying anything on hover
- Visual aspect not being final
- Calendar does not exist
- Missing a lot of animations
- There is currently a bug related to the selection of content (content with dropdowns once selected become all blue colored)

### CODE REVIEW

File: frontend/src/components/Portals/ScheduleModal.jsx [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+                                        </div>
+                                    </div>
+                                    )}
+                                    {showContentsPortal && <ScheduleContentModal
+                                        showContentsPortal={showContentsPortal} 
+                                        setShowContentsPortal={setShowContentsPortal}
+                                        widgetId={selectedWidgetId}
+                                        contents={selectedContent}
+                                        setContents={setSelectedContent} />
+                                    }
```
Comment (fungame2270) [inline frontend/src/components/Portals/ScheduleModal.jsx]: tens o boolean fora e dentro do componente nao seria so necessario aqui fora?

File: frontend/src/pages/Schedule/index.jsx [hunk 2]
```diff
@@ -1,66 +1,66 @@
...
-                            <option selected disabled hidden>Choose here</option>
-                            {templates.length !== 0 && templates.map((template) => 
-                                <option value={template.id}>{template.name}</option>
-                            )}
-                        </select>
+                    {showPortal && <ScheduleModal
+                            showPortal={showPortal} 
+                            setShowPortal={setShowPortal}
+                            selectedGroup={groups.at(selectedGroupId)} />
+                    }
```
Comment (fungame2270) [inline frontend/src/pages/Schedule/index.jsx]: tens o boolean aqui fora e la dentro nao seria necessario so la dentro

File: frontend/src/components/Portals/ScheduleModal.jsx (hunk 1 above)
Comment (pdPinho) [reply frontend/src/components/Portals/ScheduleModal.jsx]: Penso que somente tendo isso lá dentro força o load do modal, por isso irei tirar o que está lá dentro e irei deixar o de fora. Mas vou testar isso.

File: frontend/src/pages/Schedule/index.jsx (hunk 2 above)
Comment (pdPinho) [reply frontend/src/pages/Schedule/index.jsx]: Penso que somente tendo isso lá dentro força o load do modal, por isso irei tirar o que está lá dentro e irei deixar o de fora. Mas vou testar isso.

**Review by fungame2270 — APPROVED** [review]
lgtm

**Pull request merged**